edalize/templates/vivado/vivado-program.tcl.j2

```diff
@@ -1,33 +1,44 @@
 # Auto-generated program tcl file
 
+set hw_target {{ hw_target }}
 set bit {{ bitstream_name }}
 set part {{ part }}
+set jtag_freq {{ jtag_freq }}
 
 open_hw
 connect_hw_server
 
 set found 0
 
-foreach { hw_target } [get_hw_targets] {
+if {$hw_target == ""} {
+    set hw_targets [get_hw_targets]
+} else {
+    set hw_targets [get_hw_targets $hw_target]
+}
+
```

Comment:
I think we can simplify this. If $hw_target is empty then I believe [get_hw_targets $hw_target] will just evaluate to [get_hw_targets].

Reply:
Just checked this and it didn't work.  Only took me 9 months :).